Tella/Scenes/Uwazi/ViewModel/AddTemplateVM.swift

```diff
@@ -68,9 +71,13 @@ class AddTemplateViewModel: ObservableObject {
     }
     
     func downloadTemplate(template: CollectedTemplate) {
-        var template = template
-        self.downloadTemplate(template: &template)
-        self.templateItemsViewModel.first(where: {template.templateId == $0.id})?.isDownloaded = true
+        self.isLoading = true
```

Comment:
Instead of separate completion booleans `(isLoading)`, maybe we can use `Result` type to represent success or failure with an associated value (the downloaded template). This improves clarity and error handling?

Reply:
Makes sense! That would improve the process of handling errors in the codebase. However, I would push this change to another cycle so as not to go out of the scope of this PR. What do you think @dhekra-rouatbi @ahlem-jarrar?